source/wp-content/themes/wporg-developer-2023/src/code-hooks/block.php

```diff
@@ -29,12 +29,23 @@ function init() {
  * @return string Returns the block markup.
  */
 function render() {
-	$has_hooks   = ( post_type_has_hooks_info() && ( $hooks   = get_hooks() ) && $hooks->have_posts() );
+	if ( ! post_type_has_hooks_info() ) {
+		return '';
+	}
+
+	$hooks = get_hooks();
 
-	if ( ! $has_hooks ) {
+	if ( ! $hooks->have_posts() ) {
 		return '';
 	}
 
+	$content = array();
+	while ( $hooks->have_posts() ) {
+		$hooks->the_post();
+		$content[] = do_blocks( '<!-- wp:wporg/code-reference-title {"isLink":true,"tagName":"div","fontSize":"normal", "style":{"spacing":{"margin":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|20"}}}} /-->' );
+		wp_reset_postdata();
+	}
+
```

Comment:
The post data doesn't need to be reset with every iteration, you can run `wp_reset_postdata` after the loop.

```suggestion
	}
	wp_reset_postdata();

```